terraform/modules/gluekube/node.tf

```diff
@@ -0,0 +1,53 @@
+
+resource "hcloud_server" "cluster_node" {
+  for_each    = toset([for i in range(0, var.node_count) : tostring(i)])
+  name        = "${var.name}-${each.key}"
+  image       = var.image
+  server_type = var.instance_type
+  location    = var.region
+  public_net {
+    ipv4_enabled = true
+    ipv6_enabled = true
+  }
+  user_data = base64encode("${templatefile("${path.module}/cloudinit/cloud-init-${var.role}.yaml",{
+    public_key = autoglue_ssh_key.ssh-key.public_key
+    hostname = "${var.role}-${var.name}-${each.key}"
+  })}")
+
+  firewall_ids = [hcloud_firewall.firewall.id]
+}
+
+resource "hcloud_server_network" "cluster_node_network" {
+  for_each = hcloud_server.cluster_node
+  server_id = each.value.id
+  subnet_id = var.subnet_id
+}
+
+
+
+resource "hcloud_firewall" "firewall" {
```

Comment:
can we have two rules? One that allows all IPV4 from private networks and one that allows all public traffic to the necessary ports, I think it's just TCP 22, 443, and 80?

Outgoing should be open for all.